src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -133,12 +133,21 @@ variant_add_commit_details (GVariantDict *dict,
 {
   g_autoptr(GVariant) metadata = NULL;
   g_autofree gchar *version_commit = NULL;
+  g_autofree gchar *end_of_life = NULL;
   guint64 timestamp = 0;
 
   timestamp = ostree_commit_get_timestamp (commit);
   metadata = g_variant_get_child_value (commit, 0);
   if (metadata != NULL)
-    g_variant_lookup (metadata, "version", "s", &version_commit);
+    {
+      g_variant_lookup (metadata, "version", "s", &version_commit);
+      /* avoid double insertion */
+      if (prefix != NULL)
+        g_variant_lookup(metadata, "end-of-life", "s", &end_of_life);
```

Comment:
Also, for ostree metadata there's a convention to use namespace prefixes; for example, most of the metadata that rpm-ostree uses starts with `rpmostree.`, like `rpmostree.inputhash`.  The`version` metadata key should probably have been `ostree.version`.

I think we should have this metadata defined in ostree, so it'd be `ostree.end-of-life`.  In fact...this is even an example in ostree today:

```
$ git describe --tags 
v2017.8-28-ge87102b7
$ git grep end-of-life
man/ostree-summary.xml:                  <command>exampleos.end-of-life "@t 1445385600"</command>.
src/libostree/ostree-repo.h: * `exampleos.end-of-life`. The `ostree.` prefix is reserved.
```

Maybe we make a document upstream that has well-known metadata keys?